frontend/src/metabase/admin/datamodel/components/database/MetricsList.jsx

```diff
@@ -3,20 +3,15 @@ import PropTypes from "prop-types";
 import { Link } from "react-router";
 import { t } from "ttag";
 import MetricItem from "./MetricItem.jsx";
+import Metrics from "metabase/entities/metrics";
 
+@Metrics.loadList({ wrapped: true })
 export default class MetricsList extends Component {
-  static propTypes = {
-    tableMetadata: PropTypes.object.isRequired,
-    onRetire: PropTypes.func.isRequired,
-  };
+  static propTypes = { tableMetadata: PropTypes.object.isRequired };
 
   render() {
-    let { tableMetadata } = this.props;
-
-    tableMetadata.metrics = tableMetadata.metrics || [];
-    tableMetadata.metrics = tableMetadata.metrics.filter(
-      mtrc => mtrc.archived === false,
-    );
+    const { metrics: allMetrics, tableMetadata } = this.props;
+    const metrics = allMetrics.filter(m => m.table_id === tableMetadata.id);
```

Comment:
I wasn't sure how to use the `query` option to `loadList`, so I just filtered on the client. I imagine the cardinality is small enough that this will work, but it feels cleaner to use `query` IMO. Is there anywhere I can look to as an example?

Reply:
`query` is an object (or function, same as `id` for the object loaders) that's serialized to a query string and passed along in the request, so that functionality depends on what's implemented in the API endpoint.

Here's an example https://github.com/metabase/metabase/blob/master/frontend/src/metabase/components/CollectionLanding.jsx#L99

Occasionally we implement a custom `list` like this https://github.com/metabase/metabase/blob/master/frontend/src/metabase/entities/tables.js#L36 which delegates to different real endpoints, in order to work nicely with entities, but that's a bit of a hack.

One thing to be aware of with the entity loaders is some of our endpoints return nested data which might make loading data in sub-component like `MetricsList` unnecessary. Currently the data model loads `/api/database/:id/metadata` which includes every table and every table's metrics, segments, and fields. I don't think we should be loading every table up front, but since you can see all metrics, segments, and fields for a table in the table page it might make sense to continue loading those together.

We don't really have a great way to use the entity loaders (to get features like `wrapped`) in the case where the data has been loaded by a different entity (although for `EntityObjectLoader` you can pass an array of string to the `properties` prop, which bypasses loading if the object has been loaded with all specified properties, e.x. in `EntityName`, but we don't have anything like that for lists)

I don't think this is a big deal especially for things like the admin panel, and in some ways it's an improvement since it should cache the results.